02-peer-protocol.md

```diff
@@ -496,14 +498,15 @@ A sending node:
 
 A receiving node:
   - if it hasn't received a `funding_signed` (if it is a funder) or a `funding_created` (if it is a fundee):
-    - SHOULD fail the connection
+    - SHOULD send a `warning`.
   - if the `scriptpubkey` is not in one of the above forms:
-    - SHOULD fail the connection.
+    - SHOULD send a `warning`.
   - if it hasn't sent a `funding_locked` yet:
     - MAY reply to a `shutdown` message with a `shutdown`
   - once there are no outstanding updates on the peer, UNLESS it has already sent a `shutdown`:
     - MUST reply to a `shutdown` message with a `shutdown`
   - if both nodes advertised the `option_upfront_shutdown_script` feature, and the receiving node received a non-zero-length `shutdown_scriptpubkey` in `open_channel` or `accept_channel`, and that `shutdown_scriptpubkey` is not equal to `scriptpubkey`:
+    - MAY send a `warning`.
```

Comment:
Why?  I mean, you *can*, but if they screw up it seems a bit harsh.  Dropping the connection means you ignore it, and they can try again.

Reply:
I don't see how its "harsh"? Its a channel open - you can try again freely without reconnecting if you send an error, an error message is somewhat less annoying cause you don't have to do a whole P2P reconnect.

Reply:
No, this is not channel open?  (That was above, and I change it as you suggested)

 This is "you tried to close to wrong scriptpubkey".

Reply:
Ah! I think I'm blind. I guess this still falls into the same general category as the all-0 errors - it feels like yet another case of "well, if we reconnect and then retry maybe it'll resolve itself", which requires a bunch of never-used logic to ensure we fail after a few reconnect/retry loops as its otherwise unrecoverable. I guess I'm a bit confused by "MUST fail the connection" (which means force-close, no? The updates in 00 don't mention it explicitly) and also "MAY send a warning" - we should error if we fail the channel.